### PR TITLE
gargoyle-postgresql: Add gargoyle-pg-run for general purpose uses

### PR DESCRIPTION
**Pull request by 3noch**
This allows us to generalize gargoyle-database-using programs. For example:
  * `gargoyle-pg-run db psql`
  * `gargoyle-pg-run db pg_dump`
  * `gargoyle-pg-run db my_personal_script --db {}`

### CODE REVIEW

**Comment by danharaj** [comment]
is this use of `{}` common?

**Comment by 3noch** [comment]
It is. It's what `xargs` uses.

**Comment by 3noch** [comment]
It's also used by `find` for similar purpose.

**Comment by 3noch** [comment]
Another cool trick: `gargoyle-pg-run db bash -s {}` gives you an interactive shell where `$1` is the path to the open PostgreSQL unix socket.

File: gargoyle-postgresql/src-bin/gargoyle-pg-run.hs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  args <- getArgs
+  case args of
+    dbPath:cmd:cmdArgs ->
+      exitWith =<< runPgLocalWithSubstitution defaultPostgres dbPath cmd (\conn -> if null cmdArgs then [conn] else substArgs conn cmdArgs) Nothing
+    _ -> do
+      pname <- getProgName
+      putStrLn $ unwords [ "USAGE:", pname, "<path>", "<command>", "[...<arguments>]" ]
+      putStrLn "\t<path>: path to local db"
+      putStrLn "\t<command>: command to run"
+      putStrLn "\t<arguments>: list of arguments to <command> where the special argument '{}' is expandend into a connection string; if <arguments> is empty, '{}' will be supplied as the only argument by default"
```
[the remark this answers is not in the document]
Comment (3noch) [reply gargoyle-postgresql/src-bin/gargoyle-pg-run.hs]: :man_facepalming: Why u always gotta be finding my typos.

**Pull request merged**